refactor: Embrace default chaining

kriskowal opened the pull request:
Follow-up upon #1514 completion.

We left several notes to our future selves to embrace default chaining when we could.

erights commented on packages/compartment-mapper/src/bundle-lite.js:
```diff
@@ -425,13 +425,7 @@ export const makeFunctorFromMap = async (
+          const module = modulesByKey[aliases.get(key) ?? key];
           if (module === undefined) {
             throw new Error(
               `Unable to locate module for key ${q(key)} import specifier ${q(
```
Not quite equivalent because the before code might use the value of `alias` in the error message and the after code will only ever use `key`. Aside from syntax, which is the better behavior?

kriskowal replied:
Neither. Posting an update that varies the error depending on whether an alias was followed to provide a better debugging experience.

erights commented on packages/compartment-mapper/src/bundle.js:
```diff
@@ -432,13 +432,8 @@ export const makeFunctorFromMap = async (
+          const module = modulesByKey[key] ?? modulesByKey[importSpecifier];
           if (module === undefined) {
             throw new Error(
               `Unable to locate module for key ${q(key)} import specifier ${q(
```
Same question re error message

kriskowal reviewed (commented):
Will my apologies for overwriting the commit rather than presenting a diff, I’ve posted an update with a more nuanced take for better error messages.

And also rediscovered that there is duplicate code that should have disappeared in factoring bundle-lite out of bundle, which I will address in another change.

erights reviewed (approved):
LGTM

kriskowal committed:
refactor: Embrace default chaining

Follow-up upon #1514 completion.

kriskowal committed:
chore: Add patch changesets for default-chaining refactor

Address judge round-1 summary-fix: the refactor touches two published
packages (@endo/captp and @endo/compartment-mapper) and warrants a
patch changeset per package. The compartment-mapper changeset notes
the diagnostic split (alias-resolved vs direct-key error messages),
which is technically observable to consumers who string-match error
text.
